operator/plugin/config.go

```diff
@@ -0,0 +1,110 @@
+package plugin
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/observiq/stanza/operator"
+	"github.com/observiq/stanza/operator/helper"
+	"github.com/observiq/stanza/pipeline"
+	yaml "gopkg.in/yaml.v2"
+)
+
+var _ operator.MultiBuilder = (*Config)(nil)
+
+type Config struct {
+	helper.WriterConfig
+	plugin     *Plugin
+	Parameters map[string]interface{} `json:",squash" yaml:",squash"`
+}
+
+func (c *Config) BuildMulti(bc operator.BuildContext) ([]operator.Operator, error) {
+	params := c.getRenderParams(bc)
+	pipelineConfigBytes, err := c.plugin.Render(params)
+	if err != nil {
+		return nil, err
+	}
+
+	var pipelineConfig struct {
+		Pipeline pipeline.Config
+	}
+	if err := yaml.Unmarshal(pipelineConfigBytes, &pipelineConfig); err != nil {
+		return nil, err
+	}
+
+	nbc := bc.WithSubNamespace(c.ID())
+	return pipelineConfig.Pipeline.BuildOperators(nbc)
+}
+
+func (c *Config) getRenderParams(bc operator.BuildContext) map[string]interface{} {
+	// Copy the parameters to avoid mutating them
+	params := map[string]interface{}{}
+	for k, v := range c.Parameters {
+		params[k] = v
+	}
+
+	// Add ID and output to params
+	params["input"] = bc.PrependNamespace(c.ID())
+	params["output"] = c.yamlOutputs(bc)
+	return params
+}
+
+func (c *Config) yamlOutputs(bc operator.BuildContext) string {
+	outputIDs := c.OutputIDs
+	if len(outputIDs) == 0 {
+		outputIDs = bc.DefaultOutputIDs
+	}
+	namespacedOutputs := make([]string, 0, len(outputIDs))
+	for _, outputID := range outputIDs {
+		namespacedOutputs = append(namespacedOutputs, bc.PrependNamespace(outputID))
+	}
+	return fmt.Sprintf("[%s]", strings.Join(namespacedOutputs, ","))
+}
+
+func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var m map[string]interface{}
+	if err := unmarshal(&m); err != nil {
+		return err
+	}
+
+	if id, ok := m["id"]; ok {
+		if idString, ok := id.(string); ok {
+			c.OperatorID = idString
+			delete(m, "id")
+		}
+	}
+
+	if t, ok := m["type"]; ok {
+		if typeString, ok := t.(string); ok {
+			c.OperatorType = typeString
+			delete(m, "type")
+		} else {
+			return fmt.Errorf("invalid type %T for operator type", t)
+		}
+	} else {
+		return fmt.Errorf("missing required field 'type'")
+	}
+
+	if output, ok := m["output"]; ok {
+		outputIDs, err := helper.NewOutputIDsFromInterface(output)
+		if err != nil {
+			return err
+		}
+		c.OutputIDs = outputIDs
+		delete(m, "output")
+	}
+
+	c.Parameters = m
+	return nil
+}
+
+func (c Config) MarshalYAML() (interface{}, error) {
```

Comment:
Comment

Reply:
Would you mind looking into our linting config? Seems like this kind of thing should be called out automatically.

Reply:
Yeah -- our current linting does not enforce comments. Additionally, we do not currently enforce linting (it's just used by editors or `make lint`). I'll see if I can change at least the first part -- we should probably talk more about automatic enforcement

Reply:
There are a _lot_ of things that are not in line with `golint`'s rules. We should probably scan through them at some point and decide how we want to handle linting moving forward. I'll fix the missing comments on this PR, but am going to skip the rest automatic linting for now

Reply:
I created #146 to track this.